App/Controllers/RoleController.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+namespace App\Controllers;
+
+use Core\View;
+use App\Models\Role;
+use App\Models\Permission;
+
+class RoleController {
+
+  public function index()
+  {
+      $query = Role::getDB();
+      $roles = $query->readAll(Role::getTableName(),Role::class);
+
+      View::render("roles/index.view.php",["roles" => $roles]);
+  }
+
+  public function create()
+  {
+      View::render("roles/create.view.php");
+  }
+
+  public function edit($id)
+  {
+      if (isset($id))
+      {
+          $role = Role::getDB()->readById(Role::getTableName(),$id,Role::class);
+          $sqlParams['role_id']=$id;
+          $permissions = Permission::getDB()->read(Permission::getTableName(),$sqlParams,Permission::class);
+
+          $permissionsArray=[];
+          if (count($permissions)>0)
+          {
+              foreach ($permissions as $permission)
+              {
+                  $permissionsArray[$permission->getType()] = true;
+              }
+          }
+          View::render("roles/edit.view.php",["role" => $role,"permissions" => $permissionsArray]);
+      }
+  }
+
+  public function store()
+  {
+      $role = new Role();
+      $role->setName($_POST['name']);
+
+      $permissions =[];
+      if (isset($_POST['MANAGE_ROLES']))
```

Comment:
Create an ENUM with all the available permissions eg.

```
<?php

namespace App\Enum;

class UserPermissionsEnum{
  const MANAGE_ROLES = "MANAGE_ROLES";
  const MANAGE_TEAM_LEADERS = "MANAGE_TEAM_LEADERS";
  const MANAGE_USERS = "MANAGE_USERS";
  const MANAGE_TASKS = "MANAGE_TASKS";
  const COMPLETE_TASKS = "COMPLETE_TASKS";

  /**
     * Return all above permissions;
     * @return array
     * @throws \ReflectionException
     */
    public static function getAllPermissions()
    {
        $reflectionClass = new ReflectionClass(__CLASS__);
        return array_values($reflectionClass->getConstants());
    }
}

```

After that you can come below and do this:

```
 $permissions = UserPermissionsEnum::getAllPermissions();
 foreach($_POST as $permission => $value){
   if(in_array($permissions, $permission)){
     $permissions[count($permissions)] = $this->addNewPermision($permission);
   }
 }
```